slick.core.js

```diff
@@ -96,15 +138,17 @@
      *      If not specified, the scope will be set to the <code>Event</code> instance.
      */
     this.notify = function (args, e, scope) {
-      e = e || new EventData();
+
+      if(!(e instanceof EventData))
+        e = new EventData(e, args);
       scope = scope || this;
 
-      var returnValue;
       for (var i = 0; i < handlers.length && !(e.isPropagationStopped() || e.isImmediatePropagationStopped()); i++) {
-        returnValue = handlers[i].call(scope, e, args);
+        const returnValue = handlers[i].call(scope, e, args);
+        e.addReturnValue(returnValue);
       }
 
-      return returnValue;
+      return e;
```

Comment:
if I wonder if we could simplify this to return an object as `{ event, value }` instead of having to call `getReturnValue()`, this way we would get both in the same return. It could also be changed in a future PR